### PR TITLE
refactor(store): finish migration to redux-toolkit

### PR DESCRIPTION
**Pull request by dubisdev**
This PR continues #594 

The Tasks store has been migrated to RTK and now all the codebase is using redux toolkit

### CODE REVIEW

**Review by sekwah41 — CHANGES_REQUESTED** [review]
Left a response on discord. Seems to be a strange issue with skipping tasks on the timer window with this pr or the previous one

File: app/renderer/src/store/tasks/index.ts [hunk 1]
```diff
@@ -1,3 +1,322 @@
...
+
+      const firstNotDoneCard = notDoneCards.at(0);
+
+      if (!firstNotDoneCard) return;
+
+      state[listIndex] = {
+        ...list,
+        cards: [
+          // Not done without the old first card
+          ...notDoneCards.splice(0, 1),
```
Comment (dubisdev) [inline app/renderer/src/store/tasks/index.ts:237]: Here is the error

splice returns the deleted one and not the rest of the Tasks

**Comment by dubisdev** [comment]
I think the problem is solved now :)

**Comment by sekwah41** [comment]
Looks all good :D

**Review by sekwah41 — APPROVED** [review]

**Pull request merged**